From: Jochen Sprickerhof <jspricke@debian.org>
Date: Wed, 18 Oct 2023 23:16:28 +0200
Subject: Fix test

---
 tests/data/test1538 | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/data/test1538 b/tests/data/test1538
index 59cd162..d5ae382 100644
--- a/tests/data/test1538
+++ b/tests/data/test1538
@@ -132,7 +132,8 @@ e96: QUIC connection error
 e97: proxy handshake error
 e98: SSL Client Certificate required
 e99: Unrecoverable error in select/poll
-e100: Unknown error
+e100: ECH attempted but failed
+e101: Unknown error
 m-1: Please call curl_multi_perform() soon
 m0: No error
 m1: Invalid multi handle
