From: Stephen Farrell <stephen.farrell@cs.tcd.ie>
Date: Wed, 4 Oct 2023 02:45:18 +0100
Subject: spacing

---
 lib/vtls/openssl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/vtls/openssl.c b/lib/vtls/openssl.c
index 9c1a8b6..41da90c 100644
--- a/lib/vtls/openssl.c
+++ b/lib/vtls/openssl.c
@@ -4111,7 +4111,7 @@ static CURLcode ossl_connect_step2(struct Curl_cfilter *cf,
         size_t out_name_len = 0;
 
         SSL_get0_ech_name_override(backend->handle, &out_name, &out_name_len);
-        if (out_name_len > 0) 
+        if(out_name_len > 0)
           infof(data, "ECH: outername is %s", out_name);
 
         SSL_get0_ech_retry_configs(backend->handle, &rcs, &rcl);
